sdk/hash/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+#![no_std]
+#![cfg_attr(RUSTC_WITH_SPECIALIZATION, feature(min_specialization))]
+#[cfg(feature = "borsh")]
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+#[cfg(any(feature = "std", target_arch = "wasm32"))]
+extern crate std;
+#[cfg(feature = "bytemuck")]
+use bytemuck_derive::{Pod, Zeroable};
+#[cfg(feature = "serde")]
+use serde_derive::{Deserialize, Serialize};
+#[cfg(any(all(feature = "borsh", feature = "std"), target_arch = "wasm32"))]
+use std::string::ToString;
+use {
+    core::{
+        convert::TryFrom,
+        fmt, mem,
+        str::{from_utf8, FromStr},
+    },
+    solana_sanitize::Sanitize,
+};
+#[cfg(target_arch = "wasm32")]
+use {
+    js_sys::{Array, Uint8Array},
+    std::{boxed::Box, format, string::String, vec},
+    wasm_bindgen::{prelude::*, JsCast},
+};
+
+/// Size of a hash in bytes.
+pub const HASH_BYTES: usize = 32;
+/// Maximum string length of a base58 encoded hash.
+pub const MAX_BASE58_LEN: usize = 44;
+
+/// A hash; the 32-byte output of a hashing algorithm.
+///
+/// This struct is used most often in `solana-sdk` and related crates to contain
+/// a [SHA-256] hash, but may instead contain a [blake3] hash, as created by the
+/// [`blake3`] module (and used in [`Message::hash`]).
+///
+/// [SHA-256]: https://en.wikipedia.org/wiki/SHA-2
+/// [blake3]: https://github.com/BLAKE3-team/BLAKE3
+/// [`blake3`]: crate::blake3
+/// [`Message::hash`]: crate::message::Message::hash
```

Comment:
nit: sorry I only just noticed this, but the links don't make sense anymore, so best to remove the last bit of the sentence
```suggestion
/// This struct is used most often in `solana-sdk` and related crates to contain
/// a [SHA-256] hash, but may instead contain a [blake3] hash.
///
/// [SHA-256]: https://en.wikipedia.org/wiki/SHA-2
/// [blake3]: https://github.com/BLAKE3-team/BLAKE3
```